### DISCUSSION TITLE
How to pass started state ?

### DISCUSSION THREAD

**Discussion by evdevru** [jeremyckahn/shifty#141] [Q&A]
I have counter from 1 to 100. I want to start from easing on state 25 with expired time. I dont no how to calculate executed time if easing is not linear. How can i do it ?

**Comment by jeremyckahn** [comment]
Hi @evdevru, I think the [`seek`](https://jeremyckahn.github.io/shifty/doc/Tweenable.html#seek) method should serve your need. I put together a CodePen to show how it might work: https://codepen.io/jeremyckahn/pen/wvpgoVe?editors=1011

```js
const output = document.getElementById("output");
const duration = 5000;

const tweenable = tween({
  from: { number: 0 },
  to: { number: 100 },
  duration,
  render: ({ number }) => {
    output.innerHTML = Math.round(number);
  }
});

tweenable.seek(0.25 * duration);
```

The thing to keep in mind is that you'll need to call `seek` right after starting the tween. Shifty should calculate all of the timing and easing offsets for you. Let me know if this helps!

**Reply by evdevru** [reply]
Thank you for response! But its working only for linear easing (as i write later). If you use easing: 'easeInQuad', get 6.25 number, thats incorrect.

If use seek (or negative delay) need formula for easing, but i dont no them.

**Reply by jeremyckahn** [reply]
> If you use easing: 'easeInQuad', get 6.25 number, thats incorrect.

`6.25` is the correct value for 25% of an `easeInQuad` tween. Here's the source for that easing: https://github.com/jeremyckahn/shifty/blob/8bfdceb45d61ab88bceab4a75c34a2758574eda5/src/easing-functions.js#L56

We can take this and plug in `.25` for `pos`:

```js
Math.pow(.25, 2) // 0.0625
```

Which is equivalent to 6.25 when it gets rendered. It seems that Shifty is working correctly here. Perhaps something else in your project needs to change in order to achieve the desired effect?

**Reply by evdevru** [reply]
Please, can correct me ? May be i dont understand

Example:

duration: 120000
easing: 'easeInQuad'
from: 1
to: 100
current: 2

In that easing 2 = ~12060 milliseconds (get using render)

2 = 2% of range 100
2% of 120000 (duration) = 2400 milliseconds, but i want to get 12060

In linear is working, but how to adopt to others

**Reply by jeremyckahn** [reply]
I'm not able to understand the issue given what's shown here. @evdevru would you be able to share a CodePen or something similar that demonstrates the problematic behavior?

**Reply by evdevru** [reply]
> 

It's not problem with shifty, its question, how to start from needed state using easing. Check my comments: https://codepen.io/evdevru/pen/WNdRXPY?editors=1011

**Reply by jeremyckahn** [reply]
I think I might be understanding the issue here. It seems that your current code is trying to start from a static linear value, but that would only work for a `linear` easing. [Shifty exposes the easing formulas as functions](https://jeremyckahn.github.io/shifty/doc/Tweenable.html#.formulas), so perhaps given your example you could have something like:

```js
const startingPoint = Tweenable.formulas.easeInQuad((started / 100) * duration)
```

You'd probably have to do some more transformations to the number from there, but that might help to produce a proportional starting value.
